Создан интерфейс и реализована его работоспособность.

Sashok007100 opened the pull request:
Была создана новая ветка реализации нового функционала приложения.
Был создан интерфейс в приложении на основе макета из Figma и реализована его работа на основе московых данных, которая в зависимости от правильности ответа красит рамку фотографии в красный или зелёный цвет, показывает следующий вопрос и по окончанию игры показывает результат правильных ответов и предлагает сыграть ещё раз.

Sashok007100 committed:
Были установлены шрифты, которые должны использоваться в приложении, а также цвета.
Был оформлен экран загрузки в соответствии с макетом приложения.
Были созданы и расположены UI элементы на главном экране, и настроенны в соответствии с макетом приложения.

Sashok007100 committed:
Был создан массив с вопросами.
Были созданы связи UI элементов с кодом.

Sashok007100 committed:
Была добавлена функциональность, а именно функция convert для конвертации массива вопросов в вьюмодель для корректного отображения данных ан экране. Так же была добавлена функция show для отображения данных на экране приложения.
Функции были добавлены в функцию viewDidLoad, для того, чтобы при запуске приложения подставлялись данные на экран.

Sashok007100 committed:
Была создана функция showAnswerResult, которая принимает в качестве параметра нулевое значение и в зависимости от того, правильно ли ответил пользователь на вопрос обводит рамку вокруг картинки в красный или зелёный цвета.

Sashok007100 committed:
Была создана функция showNextQuestionOrResult, которая при достижении последнего вопроса сбрасывает счётчик вопросов до первого вопроса, или же если это не последний вопрос, то увеличивает счётчик вопросов и отображает следующий вопрос.
Также теперь теперь если пользователь ответить правильно на вопрос, то счётчик правильных вопросов увеличится.

Sashok007100 committed:
Была добавлена функция show, которая принимает в качестве параметра вьюмодель результата и выводит информацию на экран в виде алерта. Также алерт в данной функции show выводит результат игры и предлагает пользователю сыграть ещё раз, сбрасывая прогресс дло первого вопроса.

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,164 @@
 import UIKit
 
+struct QuizStepViewModel {
+    let image: UIImage
+    let question: String
+    let questionNumber: String
+}
+
+struct QuizResultsViewModel {
+    let title: String
+    let text: String
+    let buttonText: String
+}
+
+struct QuizQuestion {
+    let image: String
+    let text: String
+    let correctAnswer: Bool
+}
```
Можно создать отдельную папку **Models** и вынести эти модели в разные файлы в эту папку, так нам будет проще находить их.

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,164 @@
+
 final class MovieQuizViewController: UIViewController {
+    
+    @IBOutlet weak private var imageView: UIImageView!
+    @IBOutlet weak private var textLabel: UILabel!
+    @IBOutlet weak private var counterLabel: UILabel!
+    
+    private let question: [QuizQuestion] = [
```
Круто, что ты не забываешь добавлять свойствам  и методам _private_!

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,164 @@
+        alert.addAction(action)
+        
+        self.present(alert, animated: true, completion: nil)
     }
+    
 }
```
Пустые строки в конце классов/методов можно удалять.
**SwiftLint** (сервис для чистого кода), например, может ругаться на такие места (оставлять Warning)

Nikitaaeee commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -1,10 +1,164 @@
+                correctAnswer: false),
+            QuizQuestion(
+                image: "Tesla",
+                text: "Рейтинг этого фильма больше чем 6?",
+                correctAnswer: false),
+            QuizQuestion(
+                image: "Vivarium",
+                text: "Рейтинг этого фильма больше чем 6?",
+                correctAnswer: false)
+    ]
+    
+    private var currentQuestionIndex = 0
+    private var correctAnswer = 0
+    
     // MARK: - Lifecycle
     override func viewDidLoad() {
         super.viewDidLoad()
+        
+        let currentQuestion = question[currentQuestionIndex]
+        let convertQuestion = convert(model: currentQuestion)
+        show(quiz: convertQuestion)
+    }
+    
+    @IBAction private func yesButtonClicked(_ sender: Any) {
+        let currentQuestion = question[currentQuestionIndex]
+        let givenAnswer = true
+        
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: Any) {
+        let currentQuestion = question[currentQuestionIndex]
+        let givenAnswer = false
+        
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        let questionStep = QuizStepViewModel(image: UIImage(named: model.image) ?? UIImage(),
+                                             question: model.text,
```
Круто, что ты используешь комментарии типа `// MARK: - ...` 
Можно доабавлять их и к других блокам кода, подробнее можно почитать [тут](https://habr.com/ru/articles/744470/).

Nikitaaeee commented:
Здорово, что в коммитах ты описываешь что в них было сделано!

Sashok007100 committed:
Были исправленны критичсеские замечания, а именно было испрвалено что после запуска края постера не были скрыглены, а после ответа на вопрос скругление было не достаточным и толщина рамки была неправильной толщины. Также после ответа рамка не пропадала.

Sashok007100 committed:
Была доработана работа кнопок. Теперь после того как пользователь нажмёт кнопку, она становится неактивной до смены вопроса.

Sashok007100 committed:
Были добавлены комментарии типа // MARK: - ... .
Модели были вынесены из MovieQuizViewController в отдельные файлы, которые теперь хранятся в отдельной папке Models.

Sashok007100 committed:
Были добавлены комментарии типа // MARK: - ... .
Модели были вынесены из MovieQuizViewController в отедльные файлы в папку Models.
